controllers/awsmachine_controller.go

```diff
@@ -200,10 +202,16 @@ func (r *AWSMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	infrav1.SetDefaults_AWSMachineSpec(&awsMachine.Spec)
 
+	cp, err := r.getControlPlane(ctx, log, cluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
It seems we're missing that the control plane is an optional reference, and this is going to panic when nil

Reply:
Potentially could be solved with #4817.